src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -133,6 +134,16 @@ public sealed class NodeHandler : Handler, INodeHandler
                                         cancellationToken: ExecutionContext.CancellationToken);
         }
 
+        protected override void AddEnvironmentVariable(string key, string value)
```

Comment:
why only do for node handler?
if this is a real limitation, what about other handler?

Reply:
We have observed this issue manifests in node.exe tasks only. The powershell tasks were able to deal with environment variable of higher size without fail. Hence, logging this warning for node handler only.

Reply:
@SatinsMs, i am confused, where is the limitation? Windows or Node?

Reply:
From my observation its in Node.

Reply:
Seems like a limit on windows, since the following C# code will throw exception, i am suppress the agent doesn't throw exception.
```C#
StringBuilder stringBuilder = new StringBuilder();
for (int i = 0; i < 32767; i++)
{
      stringBuilder.Append("e");
}
string longenvvalue = stringBuilder.ToString();
Environment.SetEnvironmentVariable("longenv", longenvvalue);
```

Reply:
AddEnvironmentVariable doesn't really add to the environment, rather it stores the value in a dictionary named environment which is later passed on, in processstartinfo.

Reply:
Ok, i am able to repro this problem with following C# code.
```
            StringBuilder stringBuilder = new StringBuilder();
            for (int i = 0; i < 16383; i++)
            {
                stringBuilder.Append("e");
            }
            for (int i = 0; i < 16383; i++)
            {
                stringBuilder.Append("d");
            }
            stringBuilder.Append("f");
            string longenvvalue = stringBuilder.ToString();
            Console.WriteLine(longenvvalue.Length);

            //ProcessStartInfo ps = new ProcessStartInfo("node", "-e console.log(process.env['longenv'])");
            ProcessStartInfo ps = new ProcessStartInfo("powershell", "-command \"Write-Host $env:longenv\"");
            ps.Environment["longenv"] = longenvvalue;
            ps.RedirectStandardOutput = true;
            ps.RedirectStandardError = true;
            ps.RedirectStandardError = true;
            ps.UseShellExecute = false;

            Process p = Process.Start(ps);
            string stdout = p.StandardOutput.ReadToEnd();
            Console.WriteLine(stdout.Length);
            Console.WriteLine(p.StandardError.ReadToEnd());
            p.WaitForExit();
```

Reply:
Done, as suggested by you.
Started warning about the oversized environment variables in the base handler.